packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -0,0 +1,131 @@
+  // them functionally equivalent. Only one of onAttachedToEngine or registerWith will be called
+  // depending on the user's project. onAttachedToEngine or registerWith must both be defined
+  // in the same class.
+  companion object {
+    var lastAuthorizationRequestResult: Result? = null
+    var triggerMainActivityToHideChromeCustomTab : (() -> Unit)? = null
+
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "com.aboutyou.dart_packages.sign_in_with_apple")
+      channel.setMethodCallHandler(SignInWithApplePlugin())
+    }
+  }
+
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    when (call.method) {
+      "isAvailable" -> result.success(true)
+      "performAuthorizationRequest" -> {
+        val _activity = activity
+
+        if (_activity == null) {
+          result.error("MISSING_ACTIVITY", "Plugin is not attached to an activity", call.arguments)
+          return
+        }
+
+        val url: String? = call.argument("url")
+
+        if (url == null) {
+          result.error("MISSING_ARG", "Missing 'url' argument", call.arguments)
+          return
+        }
+
+        SignInWithApplePlugin.lastAuthorizationRequestResult = result
+        SignInWithApplePlugin.triggerMainActivityToHideChromeCustomTab = {
+          val notificationIntent = _activity.getPackageManager().getLaunchIntentForPackage(_activity.getPackageName());
+          notificationIntent.setPackage(null)
+          notificationIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
+          _activity.startActivity(notificationIntent)
+        }
+
+
+        val builder = CustomTabsIntent.Builder();
+        val customTabsIntent = builder.build();
+        customTabsIntent.intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+        customTabsIntent.launchUrl(_activity, Uri.parse(url));
+      }
+      else -> {
+        result.notImplemented()
+      }
+    }
+  }
+
+  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+    activity = binding.activity
+  }
+
+  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+    onAttachedToActivity(binding)
+  }
+
+  override fun onDetachedFromActivityForConfigChanges() {
+    onDetachedFromActivity()
+  }
+
+  override fun onDetachedFromActivity() {
+    activity = null
+  }
+}
+
+/**
+ * Activity which is used when the web-based authentication flow links back to the app
+ *
+ * DO NOT rename this or it's package name as it's configured in the consumer's `AndroidManifest.xml`
+ *
+ */
+public class SignInWithAppleCallback: Activity {
+  constructor() : super()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    SignInWithApplePlugin.lastAuthorizationRequestResult!!.success(intent?.data?.toString())
```

Comment:
I think it would be safer to check here that this is not `null` just so we don't crash here. And in case it is `null`, we should just log some error message

Reply:
Crashing is fine I think (didn't bring down the app, ran into this a few times while the CCT wasn't getting closed and hence firing the callback multiple times), and then there is at least a chance it will be discovered / reported.

Just logging doesn't seem enough / wouldn't notify anyone.

But we can still check the values properly and throw a more descriptive exception.